Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -520,7 +520,15 @@ public final class PackageBuilder {
                 if subpath == "" || subpath == "." {
                     return packagePath
                 }
-                let path = packagePath.appending(RelativePath(subpath))
+
+                let path: AbsolutePath
+
+                if subpath.hasPrefix("/") {
+                    path = AbsolutePath(subpath)
+                } else {
+                    path = AbsolutePath(subpath, relativeTo: packagePath)
+                }
```

Comment:
That looks like a good method to have, but in this specific case, we do [check](https://github.com/apple/swift-package-manager/blob/95b41e9a45352dafb91fb264d31ff5024ba803aa/Sources/PackageLoading/PackageBuilder.swift#L537) if the directory exists before constructing the targets. So, I think a new `check` method will be a redundant check.